kit/src/main/java/com/oracle/javafx/scenebuilder/kit/i18n/spi/I18NResourcesProviderImpl.java

```diff
@@ -29,26 +29,32 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-
-package com.oracle.javafx.scenebuilder.kit.i18n;
+package com.oracle.javafx.scenebuilder.kit.i18n.spi;
 
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Locale;
 import java.util.PropertyResourceBundle;
 import java.util.ResourceBundle;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import java.util.spi.AbstractResourceBundleProvider;
+
+public class I18NResourcesProviderImpl extends AbstractResourceBundleProvider implements I18NResourcesProvider {
 
-public class I18NControl extends ResourceBundle.Control {
     @Override
-    public ResourceBundle newBundle(String baseName, Locale locale, String format, ClassLoader loader, boolean reload) {
+    public ResourceBundle getBundle(String baseName, Locale locale) {
         String bundleName = toBundleName(baseName, locale);
-        String resourceName = toResourceName(bundleName, "properties");
-        try (InputStream is = loader.getResourceAsStream(resourceName);
-             InputStreamReader isr = new InputStreamReader(is, "UTF-8");
+        String resourceName = ResourceBundle.Control
```

Comment:
If possible, add the resourceName to the printed exception (L61) in case an exception occurs. That way, users might be able to easily detect what is wrong (or provide better feedback)

Reply:
done